### ISSUE TITLE
repl.py crash when using numpy

### ISSUE THREAD

**Issue by Menn1s** [bpython/bpython#937]
Bpython crashes when it tries to draw the tab completion menu  or pull up suggestions (not exactly sure). It occurs at the exact point when trying to type out an array in numpy and I hit the second item in the array:
`a = np.array([1,`
and the crash triggers on typing '2' after that.

Bpython version 0.21, python 3.8.10.

```
Traceback (most recent call last):
  File "/home/dennis/.local/bin/bpython", line 8, in <module>
    sys.exit(main())
  File "/home/dennis/.local/lib/python3.8/site-packages/bpython/curtsies.py", line 201, in main
    exit_value = repl.mainloop(True, paste)
  File "/home/dennis/.local/lib/python3.8/site-packages/bpython/curtsies.py", line 121, in mainloop
    self.process_event_and_paint(e)
  File "/home/dennis/.local/lib/python3.8/site-packages/bpython/curtsies.py", line 85, in process_event_and_paint
    self.process_event(e)
  File "/home/dennis/.local/lib/python3.8/site-packages/bpython/curtsiesfrontend/repl.py", line 615, in process_event
    return self.process_key_event(e)
  File "/home/dennis/.local/lib/python3.8/site-packages/bpython/curtsiesfrontend/repl.py", line 782, in process_key_event
    self.add_normal_character(e)
  File "/home/dennis/.local/lib/python3.8/site-packages/bpython/curtsiesfrontend/repl.py", line 1068, in add_normal_character
    self.cursor_offset += 1
  File "/home/dennis/.local/lib/python3.8/site-packages/bpython/curtsiesfrontend/repl.py", line 1780, in _set_cursor_offset
    self.update_completion()
  File "/home/dennis/.local/lib/python3.8/site-packages/bpython/curtsiesfrontend/repl.py", line 1100, in update_completion
    self.list_win_visible = self.complete(tab)
  File "/home/dennis/.local/lib/python3.8/site-packages/bpython/repl.py", line 723, in complete
    self.matches_iter.update(
  File "/home/dennis/.local/lib/python3.8/site-packages/bpython/repl.py", line 318, in update
    self.start, self.end, self.current_word = self.completer.locate(
TypeError: cannot unpack non-iterable NoneType object
```

**Comment by thomasballinger** [comment]
Thanks for reporting this.
Dup of https://github.com/bpython/bpython/issues/879 ?

**Comment by sebastinas** [comment]
This should be fixed in 0.22.
